src/EditorFeatures/CSharpTest/Classification/SemanticClassifierTests.cs

```diff
@@ -3477,6 +3477,105 @@ void M()
 Class("Regex"));
         }
 
+        [Theory]
+        [CombinatorialData]
+        public async Task TestJson1(TestHost testHost)
+        {
+            await TestAsync(
+@"
+class Program
+{
+    void Goo()
+    {
+        // lang=json
+        var r = @""[/*comment*/{ 'goo': 0, bar: -Infinity, """"baz"""": true }, new Date(), text, 'str'] // comment"";
+    }
+}",
+testHost,
+Keyword("var"),
+Json.Array("["),
```

Comment:
this is correct.  the semantic classifier doesn't classify symbol declaration names.  that's because that name can be classified perfectly just with syntax.

Reply:
@CyrusNajmabadi But perhaps the subtle bit here is _syntax_ of a embedded language is classified in the semantic classifier, regardless of whether it's really syntax or semantics?

Reply:
Ah.  Yes.  You are correct there.  We consider this a semantic classification.  Specifically because we need semantics (like type-info) just to even figure out if something is regex/json.  Also, we want to be off of the synchronous classification codepaths :)